src/rb/templates/persistence_impl.erb

```diff
@@ -209,7 +209,17 @@ public class <%= model_defn.impl_name %> extends AbstractDatabaseModel<<%= model
   @Override
   protected <%= model_defn.model_name %> instanceFromResultSet(ResultSet rs, Set<Enum> selectedFields) throws SQLException {
     boolean allFields = selectedFields == null || selectedFields.isEmpty();
-    return new <%= model_defn.model_name %>(rs.getLong("id"),
+    <% if database_defn.support_missing_id %>
+    long id;
+    try {
+      id = rs.getLong("id");
+    } catch (SQLException e) {
```

Comment:
catching this as an exception seems pretty ugly.  could we check for the column as described here?

http://stackoverflow.com/questions/3599861/how-can-i-determine-if-the-column-name-exist-in-the-resultset